common/ssl_calls.c

```diff
@@ -843,3 +843,82 @@ ssl_get_cipher_name(const struct ssl_st *ssl)
 {
     return SSL_get_cipher_name(ssl);
 }
+
+/*****************************************************************************/
+int
+ssl_get_protocols_from_string(const char *str, long *ssl_protocols)
+{
+    long protocols;
+    long bad_protocols;
+    int rv;
+
+    if ((str == NULL) || (ssl_protocols == NULL))
+    {
+        return 1;
+    }
+    rv = 0;
+    protocols = 0;
+#if defined(SSL_OP_NO_SSLv3)
+    protocols |= SSL_OP_NO_SSLv3;
+#endif
+#if defined(SSL_OP_NO_TLSv1)
+    protocols |= SSL_OP_NO_TLSv1;
+#endif
+#if defined(SSL_OP_NO_TLSv1_1)
+    protocols |= SSL_OP_NO_TLSv1_1;
+#endif
+#if defined(SSL_OP_NO_TLSv1_2)
+    protocols |= SSL_OP_NO_TLSv1_2;
+#endif
+    bad_protocols = protocols;
+    if (g_pos(str, ",TLSv1.2,") >= 0)
+    {
+#if defined(SSL_OP_NO_TLSv1_2)
+        log_message(LOG_LEVEL_DEBUG, "TLSv1.2 enabled");
+        protocols &= ~SSL_OP_NO_TLSv1_2;
+#else
+        log_message(LOG_LEVEL_DEBUG, "TLSv1.2 not enabled, not available");
```

Comment:
I think something like "TLSv1.2 enabled by config, but not supported by system OpenSSL" will describe the situation better.